libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryClientSettings.java

```diff
@@ -136,13 +140,17 @@ TelemetryClientSettings build() {
     }
   }
 
-  private OkHttpClient configureHttpClient() {
+  private OkHttpClient configureHttpClient(boolean attachment) {
```

Comment:
What are some benefits and side-effects of passing a boolean flag as opposed to passing a ref to interceptor itself? One specific suggestion here is to turn this signature to:
`private OkHttpClient configureHttpClient(@Nullable Interceptor inteceptor){}`
and check if inteceptor ref is passed to this method?

then your client method might look like: 
```
OkHttpClient getClient() {
   return configureHttpClient(new GzipRequestInterceptor());
}
```
Thoughts?